### PR TITLE
feat: upgrade to usvg/resvg 0.34

### LINKED ISSUE #205: Wrong generation into PNG(need support mask-type)
During convertation from SVG to PNG I lost some context of my image (red and grey colors):
[Source](https://github.com/dicebear/dicebear/blob/main/packages/%40dicebear/bottts-neutral/src/components/eyes.ts#L25):
 text

Target: 
text-out

On the other hand everything is ok if I generate similar image
[Source](https://github.com/dicebear/dicebear/blob/main/packages/%40dicebear/bottts-neutral/src/components/eyes.ts#L20):
text

Target:
text-out

Could you help me to identify what is a root cause of that. Is this really an issue with library or there some limitations. Thanks!

### LINKED ISSUE #256: Border render incorrectly on version `2.4.1`
Hi! @yisibl

This is satori contributor. I am going to upgrade `@resvg/resvg-js` to `2.4.1` so that `satori` can support more css feature like `mix-blend-mode`, but it cause some bug unexpectedly. I'll split them into some issues so that we can track.

This issue shows that `@resvg/resvg-js` does not render top red border correctly like before.

```html
<svg width="100" height="100" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg"><defs><clipPath id="satori_bc-id"><rect x="0" y="0" width="100" height="100"/></clipPath></defs><mask id="satori_om-id"><rect x="4" y="1" width="94" height="96" fill="#fff"/><path width="100" height="100" fill="none" stroke="#000" stroke-width="2" d="M0,0 A0,0 0 0 1 0,0 h100 A0,0 0 0 1 100,0"/><path width="100" height="100" fill="none" stroke="#000" stroke-width="4" d="M100,0 A0,0 0 0 1 100,0 v100 A0,0 0 0 1 100,100"/><path width="100" height="100" fill="none" stroke="#000" stroke-width="6" d="M100,100 A0,0 0 0 1 100,100 h-100 A0,0 0 0 1 0,100"/><path width="100" height="100" fill="none" stroke="#000" stroke-width="8" d="M0,100 A0,0 0 0 1 0,100 v-100 A0,0 0 0 1 0,0"/></mask><path width="100" height="100" clip-path="url(#satori_bc-id)" fill="none" stroke="red" stroke-width="2" d="M0,0 A0,0 0 0 1 0,0 h100 A0,0 0 0 1 100,0"/><path width="100" height="100" clip-path="url(#satori_bc-id)" fill="none" stroke="green" stroke-width="4" d="M100,0 A0,0 0 0 1 100,0 v100 A0,0 0 0 1 100,100"/><path width="100" height="100" clip-path="url(#satori_bc-id)" fill="none" stroke="blue" stroke-width="6" d="M100,100 A0,0 0 0 1 100,100 h-100 A0,0 0 0 1 0,100"/><path width="100" height="100" clip-path="url(#satori_bc-id)" fill="none" stroke="yellow" stroke-width="8" d="M0,100 A0,0 0 0 1 0,100 v-100 A0,0 0 0 1 0,0"/></svg>
```

You can see the result in playground with svg above generated by `satori`

### PR DESCRIPTION
**Pull request by zimond**
fixes #256 
closes #205

### CODE REVIEW

File: src/options.rs [hunk 1]
```diff
@@ -34,6 +32,31 @@ enum FitToDef {
...
+            width as f32 / size.width() as f32,
+            height as f32 / size.height() as f32,
+        );
+        if width == 0 || height == 0 {
+            Err(Error::ZeroSized)
+        } else {
+            Ok((width, height, transform))
+        }
+    }
+}
```
Comment (yisibl) [inline src/options.rs:58]: 由于 [resvg 0.33.0](https://github.com/RazrFalcon/resvg/blob/master/CHANGELOG.md#removed-1) 删除了 `resvg::FitTo` 方法，这里我们进行了重写，需要增加 FitTo height 的测试用例。

Comment (yisibl) [reply src/options.rs:58]: 在这里完成：https://github.com/yisibl/resvg-js/pull/259

**Comment by Jackie1210** [comment]
Just curious, why we don't upgrade to 0.35?

**Comment by zimond** [comment]
@Jackie1210 haha because 0.34 fork is prepared for quite some time. So this PR could quickly catch up with a recent version of resvg. Besides, starting from 0.33, our bbox related apis are broken due to resvg changes, so even if we upgrade to 0.35, this branch could hardly be merged any time soon.

**Comment by Jackie1210** [comment]
@zimond Thanks for your patient explanation.

**Comment by yisibl** [comment]
Two key APIs that we relied on have been removed in newer versions of `resvg`.

### resvg 0.34.0
- `usvg::NodeExt::calculate_bbox` no longer includes stroke bbox.

### resvg 0.33.0
- Removed resvg::FitTo
@zimond  has reimplemented FitTo in this PR.

Perhaps we should upgrade to `resvg` `0.33.0` just to be safe?

**Comment by Jackie1210** [comment]
https://github.com/yisibl/resvg-js/issues/256 looks fine, but seems stiil render nothing for https://github.com/yisibl/resvg-js/issues/257 @zimond

**Comment by ppjjzz** [comment]
any update?

**Comment by yisibl** [comment]
> any update?

It will be released when it is ready.

**Pull request closed without merging**